### ISSUE TITLE
 Elasticsearch fails to start on Azul Platform Prime JDK (formerly Zing) from 7.11 onwards

### ISSUE THREAD

**Issue by deepak-gittyup** [elastic/elasticsearch#91577]
### Elasticsearch Version

7.17.6

### Java Version

Zing 64-Bit Tiered VM Zing22.08.0.0+3 (build 11.0.16.1-zing_22.08.0.0-b3-product-azlinuxM-X86_64, mixed mode)

### OS Version

4.4.0-210-generic #242-Ubuntu

### Problem Description

After upgrading ES to a higher version i.e 7.17.6, ES hits NullPointerException and fails to start on Azul Platform Prime JDK (formerly Zing)

The issue reproduced even on a more recent version (relative to 7.17.6): 8.4.2

While trying to identify the breaking point, I noticed that the issue is reproducible from elasticsearch-7.11.0 onwards
Issue does not reproduce on  elasticsearch-7.10.0

### Steps to Reproduce

cd /home/user/
wget https://cdn.azul.com/zing-zvm/ZVM22.08.0.0/zing22.08.0.0-3-jdk11.0.16.1-linux_x64.tar.gz
tar -xf zing22.08.0.0-3-jdk11.0.16.1-linux_x64.tar.gz

wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-7.17.6-linux-x86_64.tar.gz
tar -xf elasticsearch-7.17.6-linux-x86_64.tar.gz
cd elasticsearch-7.17.6

ES_JAVA_HOME=/home/user/zing22.08.0.0-3-jdk11.0.16.1-linux_x64 bin/elasticsearch
```
Warning: Maximum heap size rounded up to 512 MB
Exception in thread "main" java.lang.NullPointerException
        at org.elasticsearch.tools.launchers.JvmErgonomics.tuneG1GCHeapRegion(JvmErgonomics.java:86)
        at org.elasticsearch.tools.launchers.JvmErgonomics.choose(JvmErgonomics.java:60)
        at org.elasticsearch.tools.launchers.JvmOptionsParser.jvmOptions(JvmOptionsParser.java:135)
        at org.elasticsearch.tools.launchers.JvmOptionsParser.main(JvmOptionsParser.java:86)
```

**Comment by deepak-gittyup** [comment]
The current understanding is: ES (version elasticsearch-7.11.0  onwards) assumes that the JDK being used supports G1GC
But Azul Platform Prime (formerly Zing) has its own C4 garbage collector (does not have G1GC)

**Comment by nik9000** [comment]
Zing isn't supported: https://www.elastic.co/support/matrix#matrix_jvm

**Comment by Holmistr** [comment]
_Disclaimer: I work at Azul._

For the benefit of any future readers - Azul Platform Prime (formerly Azul Zing) works seemlessly with Elasticsearch. The startup issue described here was trivial - ES startup scripts applied some G1 GC tuning flags that Prime didn't support (as it doesn't have G1 GC). When those tuning flags are removed, ES starts up fine and works without problems. We have submitted a PR to not apply the VM flags when they are not supported (https://github.com/elastic/elasticsearch/pull/93197), but the change was not accepted for the reasons mentioned in the PR.

We therefore applied the fix on Azul's side - we now accept those G1 GC flags, just simply ignore them :) 

Long story short - Elasticsearch is guaranteed (by Azul) to work fine with Azul Platform Prime.
